provider/pkg/provider/provider.go

```diff
@@ -837,46 +836,16 @@ func (p *cfnProvider) Create(ctx context.Context, req *pulumirpc.CreateRequest)
 		}
 	}
 
-	// Serialize inputs as a desired state JSON.
-	jsonBytes, err := json.Marshal(payload)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to marshal as JSON")
-	}
-	desiredState := string(jsonBytes)
-
 	// Create the resource with Cloud API.
-	clientToken := uuid.New().String()
-	glog.V(9).Infof("%s.CreateResource %q token %q state %q", label, cfType, clientToken, desiredState)
-	res, err := p.cctl.CreateResource(ctx, &cloudcontrol.CreateResourceInput{
-		ClientToken:  aws.String(clientToken),
-		TypeName:     aws.String(cfType),
-		DesiredState: aws.String(desiredState),
-		RoleArn:      p.roleArn,
-	})
-	if err != nil {
+	glog.V(9).Infof("%s.CreateResource %q", label, cfType)
+	id, resourceState, err := p.client.Create(ctx, cfType, payload)
```

Comment:
I think this error needs to be bound to a separate variable so we can inspect it later to decide if we're returning a partialCreate error – so the error doesn't get overwritten by a serialization error.

Reply:
Yes, great catch